routing/routing_session.hpp

```diff
@@ -190,40 +231,43 @@ class RoutingSession : public traffic::TrafficObserver, public traffic::TrafficC
 
     DoReadyCallback(RoutingSession & rs, ReadyCallback const & cb)
         : m_rs(rs), m_callback(cb)
-    {
-    }
+    {}
 
     void operator()(std::shared_ptr<Route> route, RouterResultCode e);
   };
 
   void AssignRoute(std::shared_ptr<Route> route, RouterResultCode e);
-
-  /// Returns a nearest speed camera record on your way and distance to it.
-  /// Returns kInvalidSpeedCameraDistance if there is no cameras on your way.
-  double GetDistanceToCurrentCamM(SpeedCameraRestriction & camera, DataSource const & dataSource);
-
   /// RemoveRoute removes m_route and resets route attributes (m_state, m_lastDistance, m_moveAwayCounter).
   void RemoveRoute();
   void RebuildRouteOnTrafficUpdate();
 
   double GetCompletionPercent() const;
   void PassCheckpoints();
 
+  void ProcessCameraWarning();
+  void PassCameraToWarned();
+
 private:
   std::unique_ptr<AsyncRouter> m_router;
   std::shared_ptr<Route> m_route;
   State m_state;
   bool m_isFollowing;
   Checkpoints m_checkpoints;
-  size_t m_lastWarnedSpeedCameraIndex;
-  SpeedCameraRestriction m_lastFoundCamera;
-  // Index of a last point on a route checked for a speed camera.
-  size_t m_lastCheckedSpeedCameraIndex;
-
-  // TODO (ldragunov) Rewrite UI interop to message queue and avoid mutable.
-  /// This field is mutable because it's modified in a constant getter. Note that the notification
-  /// about camera will be sent at most once.
-  mutable bool m_speedWarningSignal;
+
+  // Queue of speedCams, warnings about which has been pronounced.
+  std::queue<SpeedCameraOnRoute> m_warnedSpeedCameras;
+
+  // Queue of speedCams, that we have found, but they are too far, to make warning about them.
+  std::queue<SpeedCameraOnRoute> m_cachedSpeedCameras;
+
+  // Big red button about camera in user interface.
+  bool m_showWarningAboutSpeedCam = false;
+
+  // Flag of doing sound notification about camera on a way.
+  bool m_makeNotificationAboutSpeedCam = false;
+
+  // Index of a last checked route segment for a speed camera.
```

Comment:
The name does not match the documentation. Which of them is correct?

Reply:
delete comment because it copy the var name in fact